From: Liam Howlett <liam.howlett@oracle.com>
Subject: mm/mlock: drop dead code in count_mm_mlocked_page_nr()
Date: Wed, 15 Jun 2022 17:40:58 +0000

The check for mm being null has never been needed since the only caller
has always passed in current->mm.  Remove the check from
count_mm_mlocked_page_nr().

Link: https://lkml.kernel.org/r/20220615174050.738523-1-Liam.Howlett@oracle.com
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Suggested-by: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mlock.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/mm/mlock.c~mm-mlock-drop-dead-code-in-count_mm_mlocked_page_nr
+++ a/mm/mlock.c
@@ -531,14 +531,12 @@ static unsigned long count_mm_mlocked_pa
 	unsigned long end;
 	VMA_ITERATOR(vmi, mm, start);
 
-	if (mm == NULL)
-		mm = current->mm;
-
 	/* Don't overflow past ULONG_MAX */
 	if (unlikely(ULONG_MAX - len < start))
 		end = ULONG_MAX;
 	else
 		end = start + len;
+
 	for_each_vma_range(vmi, vma, end) {
 		if (vma->vm_flags & VM_LOCKED) {
 			if (start > vma->vm_start)
_
